Objects/abstract.c

```diff
@@ -2190,7 +2190,15 @@ _PySequence_IterSearch(PyObject *seq, PyObject *obj, int operation)
     it = PyObject_GetIter(seq);
     if (it == NULL) {
         if (PyErr_ExceptionMatches(PyExc_TypeError)) {
-            type_error("argument of type '%.200s' is not iterable", seq);
+            if (operation == PY_ITERSEARCH_CONTAINS) {
+                type_error(
+                    "argument of type '%T' is not a container or iterable",
```

Comment:
Ah, `type_error` already looks up the type, so we should use `%N`.

```suggestion
                    "argument of type '%N' is not a container or iterable",
```